cddl/concise-mid-tag.cddl

```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
-instance-type /= ip-addr
-
-full-address = [[+ label], ip4, ip6]
-ip4 = bstr .size 4
-ip6 = bstr .size 16
-label = bstr .size (1..63)
-
-mac-address = 0
-serial-number = 1
-euid = 2
-uuid = 3
-ip-addr = 4
-
-instance-values /= full-address
-instance-values /= uuid-bytes
-instance-values /= #6.47115(ueid) ;should define ueid-bytes type?
-instance-values /= mac-address-bytes
-instance-values /= serial-number-bytes
-; ...
+$$instance-claim-choice = (
+  mac-addr-key => mac-addr-type //
+  ip-addrs-key => one-or-many<ip-addr-type> //
+  serial-number-key => serial-number-type //
+  ueid-key => ueid //
+  uuid-key => uuid
+)
+
+mac-addr-key = 1
+ip-addrs-key = 2
+serial-number-key = 3
+ueid-key = 4
+uuid-key = 5
+
+ip-addr-type = ip4-addr-type / ip6-addr-type
+ip4-addr-type = bytes .size 4
+ip6-addr-type = bytes .size 16
+
+mac-addr-type = bytes .size 6
+
+serial-number-type = text
+
 ; Note: are there other well defined types under #6.4711 or EAT etc...
 
-element-name = (
-    ? label => text,
-    ? vendor => text, ; namespace authority designation
-    ? type => $module-type, ; Note: 'type' SHOULD include a namespace authority if 'vendor' is ommitted
-    ? model => text,
-    ? layer => int,
-    ? index => int,
-; **note: An extension point in element-name confuses verifiers' ability to match against evidence**
-)
-; label is defined in coswid schema (q: should we use a different value if the semantics differ?)
-; type is defined in coswid schema
-vendor=110
-model=111
-layer=112
-index=113
-
-element-value = (
+; Notes:
+; - vendor is the namespace authority designation
+; - type SHOULD include a namespace authority if vendor is ommitted
+; - An extension point in element-name confuses verifiers' ability to match
+;   against evidence
+; - label is defined in coswid schema (q: should we use a different value if
+;   the semantics differ?)
+; - type is defined in coswid schema
+element-name = {
+  ? label => text,
+  ? vendor => text,
+  ? type => $module-type,
+  ? model => text,
+  ? layer => int,
+  ? index => int,
+}
+
+vendor = 110
+model = 111
+layer = 112
+index = 113
+
+; Notes:
+; - do we need an extension point in element-value given there are extension
+;   points in endorsed-value and reference-value?
+element-value = {
   ? version => text,
   ? svn => int,
   ? raw-value-entry,
   ? digests => digests-entry,
   ? flags => bytes .bits operational-flags,
-;  * $$element-value-extension ; **note do we need an extension point in element-value given there are extension points in endorsed-value and reference-value?**
+}
+
+version = 0
+svn = 1
+digests = 2
+flags = 3
+
+raw-value-entry = (
+  raw-value => bstr,
+  ? raw-value-mask => bstr,
 )
-version=114
-svn=115
-digests=116
-flags=117
 
+raw-value = 4
+raw-value-mask = 5
+
+; Notes:
+; - one endorser benefits from the element-value of another endorser
+; - any-attribute is defined in CoSWID (this can be removed if we opt for using
+;   the extension point instead. NOTE(tho): agree
```

Comment:
Probably we already agreed on extension points (sockets) and actually no any-attributes at all in this scope? Have to reconfirm.